### PR TITLE
deps: upgrade eslint and tap-parser

### PR DESCRIPTION
**Pull request by targos**
Use the latest version of all our devDependencies.

##### Checklist

- [x] `npm test` passes
- [x] contribution guidelines followed [here](https://github.com/nodejs/citgm/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: .eslintrc.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@
     es6: true
 
   rules:
-    indent: [2, 2]
+    indent-legacy: [2, 2]
```
Comment (targos) [inline .eslintrc.yml]: There are many errors with the new `indent` rule

Comment (MylesBorins) [inline .eslintrc.yml]: we should tackle these is a separate PR

**Comment by targos** [comment]
CI: https://ci.nodejs.org/view/Node.js-citgm/job/citgm-continuous-integration/35/

**Comment by targos** [comment]
Landed in 8594213cb846e7a3e60789b8c4dc4ce8626ecc19

**Pull request closed without merging**